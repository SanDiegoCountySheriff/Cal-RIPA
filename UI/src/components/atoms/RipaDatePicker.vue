<template>
  <v-text-field
    type="date"
    v-model="model"
    :label="label"
    :rules="rules"
    :disabled="disabled"
    append-icon="mdi-calendar"
    clearable
  >
  </v-text-field>
</template>

<script>
export default {
  name: 'ripa-date-picker',

  data() {
    return {
      viewModel: this.value,
    }
  },

  computed: {
    model: {
      get() {
        return this.viewModel
      },
      set(newVal) {
        this.viewModel = newVal
        this.$emit('input', newVal)
      },
    },
  },

  watch: {
    value(newVal) {
      this.viewModel = newVal
    },
  },

  props: {
    value: {
      type: String,
      default: null,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    label: {
      type: String,
      default: '',
    },
    rules: {
      type: Array,
      default: () => [],
    },
  },
}
</script>

<style>
::-webkit-calendar-picker-indicator {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: auto;
  height: auto;
  color: transparent;
  background: transparent;
}
input::-webkit-date-and-time-value {
  text-align: left;
}
</style>
