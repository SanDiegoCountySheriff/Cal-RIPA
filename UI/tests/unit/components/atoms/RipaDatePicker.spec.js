import RipaDatePicker from '@/components/atoms/RipaDatePicker.vue'
import { mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Date Picker', () => {
  let vuetify
  let wrapper = null

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const factory = propsData => {
    return mount(RipaDatePicker, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit event', () => {
    wrapper = factory()
    wrapper.vm.model = 'New Value'

    expect(wrapper.emitted('input')).toBeTruthy()
    expect(wrapper.emitted('input').length).toEqual(1)
  })
})
