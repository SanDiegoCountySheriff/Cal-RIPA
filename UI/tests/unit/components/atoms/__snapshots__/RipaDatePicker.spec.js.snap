// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Date Picker should match snapshot 1`] = `
<v-text-field-stub
  appendicon="mdi-calendar"
  backgroundcolor=""
  clearable="true"
  clearicon="$clear"
  errorcount="1"
  errormessages=""
  label=""
  loaderheight="2"
  messages=""
  rules=""
  successmessages=""
  type="date"
/>
`;
